src/Adapter/DbTableGateway.php

```diff
@@ -11,20 +11,20 @@
 
 use Zend\Db\Sql\Where;
 use Zend\Db\Sql\Having;
-use Zend\Db\TableGateway\TableGateway;
+use Zend\Db\TableGateway\TableGatewayInterface;
 
 class DbTableGateway extends DbSelect
 {
     /**
      * Constructs instance.
      *
-     * @param TableGateway                      $tableGateway
+     * @param TableGatewayInterface             $tableGateway
      * @param null|Where|\Closure|string|array  $where
      * @param null|string|array                 $order
      * @param null|string|array                 $group
      * @param null|Having|\Closure|string|array $having
      */
-    public function __construct(TableGateway $tableGateway, $where = null, $order = null, $group = null, $having = null)
+    public function __construct(TableGatewayInterface $tableGateway, $where = null, $order = null, $group = null, $having = null)
```

Comment:
[`TableGatewayInterface`](https://github.com/zendframework/zend-db/blob/master/src/TableGateway/TableGatewayInterface.php) doesn't define the methods [`getSql`](https://github.com/zendframework/zend-paginator/blob/master/src/Adapter/DbTableGateway.php#L29) and [`getResultSetPrototype`](https://github.com/zendframework/zend-paginator/blob/master/src/Adapter/DbTableGateway.php#L44).